### PR TITLE
[Merged by Bors] - Warning message for missing events

### PR DESCRIPTION
**Pull request by Aceeri**
# Objective
- Reduce debugging burden when using events by telling user when they missed an event.

## Solution

### CODE REVIEW

**Comment by Aceeri** [comment]
Would be nice to have https://github.com/bevyengine/bevy/pull/5731 so we could tell the user which system is the offending reader of events.

**Comment by mockersf** [comment]
> Would be nice to have #5731 so we could tell the user which system is the offending reader of events.
> 
> Could add it as a `#[cfg(debug_assertions)]` if we are worried about performance cost of that.

with spans enabled, does the log have the system span?

File: crates/bevy_ecs/src/event.rs [hunk 1]
```diff
@@ -351,10 +359,18 @@ impl<E: Event> ManualEventReader<E> {
            + ExactSizeIterator<Item = (&'a E, EventId<E>)> {
         // if the reader has seen some of the events in a buffer, find the proper index offset.
         // otherwise read all events in the buffer
+        let missed = self.missed_events(events);
+        if missed > 0 {
+            let plural = if missed == 1 { "event" } else { "events" };
+            let type_name = std::any::type_name::<E>();
+            warn!("Missed {missed} `{type_name}` {plural}. Consider reading from the `EventReader` more often or updating the `Events` to update less. This is most likely due to run criteria/fixed timesteps or consuming events conditionally.");
```
Comment (mockersf) [inline crates/bevy_ecs/src/event.rs:366]: > updating the `Events` to update less

I'm not sure what that means

**Comment by Aceeri** [comment]
@mockersf I'm a bit confused, are we talking about the `tracing` system spans?

File: crates/bevy_ecs/src/event.rs (hunk 1 above)
Comment (cart) [reply crates/bevy_ecs/src/event.rs]: Add some clarity to this message.

**Review by cart — APPROVED** [review]
Pretty sure the tracing spans would add the appropriate system context, where relevant. Happy to adjust as needed though with followups. For now, I think this is good to go!

**Pull request closed without merging**